--- a/app/build.gradle
+++ b/app/build.gradle
@@ -26,6 +26,22 @@
             abiFilters "arm64-v8a", "x86_64"
         }
     }
+
+
+    signingConfigs {
+        testkey {
+            storeFile rootProject.file("app/aosp-testkey.p12")
+            storePassword "android"
+            keyAlias "testkey"
+            keyPassword "android"
+        }
+    }
+
+    buildTypes {
+        release {
+            signingConfig signingConfigs.testkey
+        }
+    }

     externalNativeBuild {
         cmake {
