diff --git a/src/path/canon.c b/src/path/canon.c
index 45d8e08..579647e 100644
--- a/src/path/canon.c
+++ b/src/path/canon.c
@@ -200,6 +200,7 @@ int canonicalize(Tracee *tracee, const char *user_path, bool deref_final,
 		 char guest_path[PATH_MAX], unsigned int recursion_level)
 {
 	char scratch_path[PATH_MAX];
+	char host_path[PATH_MAX];
 	Finality finality;
 	const char *cursor;
 	int status;
@@ -226,6 +227,13 @@ int canonicalize(Tracee *tracee, const char *user_path, bool deref_final,
 	else
 		strcpy(guest_path, "/");
 
+	/* Resolve bindings for the initial '/' component of user_path,
+	 * which is not handled in the loop below.  In particular
+	 * HOST_PATH extensions are called from there.  */
+	status = substitute_binding_stat(tracee, NOT_FINAL, recursion_level, guest_path, host_path);
+	if (status < 0)
+		return status;
+
 	/* Canonicalize recursely 'user_path' into 'guest_path'.  */
 	cursor = user_path;
 	finality = NOT_FINAL;
